Assets/Scripts/GUI/MultiplayerPanel.cs

```diff
@@ -0,0 +1,314 @@
+        }
+
+        private void UpdateDisplay()
+        {
+            if (m_RoomNumber) m_RoomNumber.text = "RoomName: " + data.roomName;
+            if (m_Nickname) m_Nickname.text = "Nickname: " + NickName;
+            Alerts();
+        }
+
+        private async void Connect()
+        {
+            if (MultiplayerManager.m_Instance != null)
+            {
+                await MultiplayerManager.m_Instance.Connect();
+            }
+        }
+
+        private async void JoinRoom()
+        {
+
+            if (MultiplayerManager.m_Instance != null)
+            {
+                await MultiplayerManager.m_Instance.JoinRoom(data);
+            }
+        }
+
+        private async void LeaveRoom()
+        {
+            if (MultiplayerManager.m_Instance != null)
+            {
+                await MultiplayerManager.m_Instance.LeaveRoom(false);
+            }
+        }
+
+        private async void Disconnect()
+        {
+            if (MultiplayerManager.m_Instance != null)
+            {
+                await MultiplayerManager.m_Instance.Disconnect();
+            }
+        }
+
+        private void OnStateUpdated(ConnectionState newState)
+        {
+            if (!m_State) return;
+            m_State.text = "State: " + StateToString(newState);
+            UpdateDisplay();
+        }
+
+        private string StateToString(ConnectionState newState)
+        {
+            switch (newState)
+            {
+                case ConnectionState.INITIALISING:
+                    return "Initialising";
+                case ConnectionState.INITIALIZED:
+                    return "Initialized";
+                case ConnectionState.DISCONNECTED:
+                    return "Disconnected";
+                case ConnectionState.DISCONNECTING:
+                    return "Disconnecting";
+                case ConnectionState.CONNECTING:
+                    return "Connecting";
+                case ConnectionState.AUTHENTICATING:
+                    return "Authenticating";
+                case ConnectionState.IN_LOBBY:
+                    return "In Lobby";
+                case ConnectionState.IN_ROOM:
+                    return "In Room";
+                case ConnectionState.ERROR:
+                    return "Error";
+                default:
+                    return "Unknown";
+            }
+        }
+
+        private void OnRoomOwnershipUpdated(bool isRoomOwner)
+        {
+            if (!m_RoomOwnership) return;
+            m_RoomOwnership.text = isRoomOwner ? "You are the Room Owner" : "You are not the Room Owner";
+        }
+
+        private Tuple<bool, string> CheckAdvancedModeActive()
+        {
+            if (PanelManager.m_Instance != null)
+            {
+                bool isAdvancedModeActive = PanelManager.m_Instance.AdvancedModeActive();
+                return Tuple.Create(isAdvancedModeActive, "Switch to beginner mode use Multiplayer");
```

Comment:
missing word "to" between "mode" and "use"

Reply:
fixed